tests/unit/test_textInfos.py

```diff
@@ -131,3 +130,103 @@ def test_mixedSurrogatePairsNonSurrogatesAndSingleSurrogatesBackward(self):
 		ti.move(textInfos.UNIT_CHARACTER, -1)
 		ti.expand(textInfos.UNIT_CHARACTER) # Range at a
 		self.assertEqual(ti.offsets, (0, 1)) # One offset
+
+
+class TestEndpoints(unittest.TestCase):
+
+	def test_lessThan(self):
+		obj = BasicTextProvider(text="abcdef")
+		wholeTi = obj.makeTextInfo(Offsets(0, 2))
+		subTi1 = obj.makeTextInfo(Offsets(0, 3))
+		subTi2 = obj.makeTextInfo(Offsets(2, 5))
+		self.assertTrue(wholeTi.start < wholeTi.end)
+		self.assertFalse(wholeTi.end < wholeTi.start)
+		self.assertFalse(wholeTi.start < wholeTi.start)
+		self.assertFalse(wholeTi.end < wholeTi.end)
+		self.assertFalse(wholeTi.start < subTi1.start)
+		self.assertTrue(subTi1.start < subTi2.start)
+		self.assertTrue(subTi2.start < subTi1.end)
+		self.assertFalse(subTi2.end < wholeTi.end)
+
+	def test_lessThanOrEqualTo(self):
+		obj = BasicTextProvider(text="abcdef")
+		wholeTi = obj.makeTextInfo(Offsets(0, 5))
+		subTi1 = obj.makeTextInfo(Offsets(0, 3))
+		subTi2 = obj.makeTextInfo(Offsets(2, 5))
+		self.assertTrue(wholeTi.start <= wholeTi.end)
+		self.assertFalse(wholeTi.end <= wholeTi.start)
+		self.assertTrue(wholeTi.start <= wholeTi.start)
+		self.assertTrue(wholeTi.end <= wholeTi.end)
+		self.assertTrue(wholeTi.start <= subTi1.start)
+		self.assertTrue(subTi1.start <= subTi2.start)
+		self.assertTrue(subTi2.start <= subTi1.end)
+		self.assertTrue(subTi2.end <= wholeTi.end)
```

Comment:
It might be easier to follow these tests if the checks were for all operators and a single pair:

``` python
def test_TextInfoEndpoint_largerAndSmaller():
	larger = obj.makeTextInfo(Offsets(0, 5)).end
	smaller = obj.makeTextInfo(Offsets(0, 3)).end

	assertTrue(larger > smaller)
	assertTrue(larger >= smaller)
	assertTrue(smaller < larger)
	assertTrue(smaller < larger)
	assertFalse(smaller == larger or larger == smaller)
	assertFalse(smaller > larger)
	... etc
```

Then you just need `test_TextInfoEndpoint_equalObjects`